client/src/components/Carousel.jsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import styled from 'styled-components';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      currentSlide: this.props.caroImage
+    }
+  }
+
+  leftClick() {
+    var imagesArray = this.props.imagesArray
+    var current = imagesArray.indexOf(this.state.currentSlide);
+    var last = imagesArray[imagesArray.length - 1]
+    var previous = imagesArray[current - 1]
+    if (current === 0) {
+      this.setState({
+        currentSlide: last
+      });
+    } else {
+      this.setState({
+        currentSlide: previous
+      });
+    }
+  }
+
+  rightClick() {
+    var imagesArray = this.props.imagesArray
+    var current = imagesArray.indexOf(this.state.currentSlide);
+    var first = imagesArray[0]
+    var next = imagesArray[current + 1]
+    if (current === imagesArray.length - 1) {
+      this.setState({
+        currentSlide: first
+      });
+    } else {
+      this.setState({
+        currentSlide: next
+      });
+    }
+  }
+
+  render() {
+    return (
+      <CaroCon>
+        <Picture>
+          <img id="carol-image" src={this.state.currentSlide}/>
+        </Picture>
+        <LeftButton onClick={this.leftClick.bind(this)}/>
+        <RightButton onClick={this.rightClick.bind(this)}/>
+      </CaroCon>
+    )
+  }
+}
+
+const CaroCon = styled.div`
+  display: flex;
+  vertical-align: middle;

```

Comment:
Why vertical-align vs flex box's align-items?

Reply:
I think this was me experimenting. Will fix!